src/In.ProjectEKA.HipService/Consent/ConsentController.cs

```diff
@@ -22,18 +22,28 @@ public ConsentController(IConsentRepository consentRepository)
             [FromHeader(Name = "X-ConsentManagerID")] string consentManagerId,
             [FromBody] ConsentArtefactRequest consentArtefactRequest)
         {
-            var consent = new Consent(consentArtefactRequest.ConsentDetail.ConsentId,
+            var consent = new Consent(null,
                 consentArtefactRequest.ConsentDetail,
                 consentArtefactRequest.Signature,
                 consentArtefactRequest.Status,
                 consentManagerId);
-            if (consentArtefactRequest.Status == ConsentStatus.GRANTED)
+            ;
+            switch (@consentArtefactRequest.Status)
             {
-                await consentRepository.AddAsync(consent);
-            }
-            else
-            {
-                await consentRepository.UpdateAsync(consent);
+                case ConsentStatus.GRANTED:
+                    consent.ConsentArtefactId = consentArtefactRequest.ConsentDetail.ConsentId;
+                    await consentRepository.AddAsync(consent);
+                    break;
+                case ConsentStatus.EXPIRED:
+                    consent.ConsentArtefactId = consentArtefactRequest.ConsentId;
+                    await consentRepository.UpdateAsync(consent);
+                    break;
+                case ConsentStatus.DENIED:
+                case ConsentStatus.REVOKED:
+                case ConsentStatus.REQUESTED:
+                    consent.ConsentArtefactId = consentArtefactRequest.ConsentDetail.ConsentId;
+                    await consentRepository.UpdateAsync(consent);
```

Comment:
this is wrong. You can leave it as it was with if and else. 

```
if (consentArtefactRequest.Status == ConsentStatus.GRANTED)
            {
                var consent = new Consent(consentArtefactRequest.ConsentDetail.ConsentId,
                    consentArtefactRequest.ConsentDetail,
                    consentArtefactRequest.Signature,
                    consentArtefactRequest.Status,
                    consentManagerId);
                await consentRepository.AddAsync(consent);
            }
            else
            {
                await consentRepository.UpdateAsync(consentArtefactRequest.ConsentId);
            }
            return NoContent();
```

Where as in the updateAsync Method expect only ConsentArtefactId 

```
var consentArtefact = await consentContext.ConsentArtefact.FirstOrDefaultAsync(c => c.ConsentArtefactId == consentArtefactId);
    consentArtefact.Status = consent.Status;
    await consentContext.SaveChangesAsync();
```